pkg/plugins/runtime/k8s/webhooks/resourceadmissionchecker.go

```diff
@@ -85,28 +80,33 @@ func (c *ResourceAdmissionChecker) isPrivilegedUser(allowedUsers []string, userI
 }
 
 func (c *ResourceAdmissionChecker) validateLabels(r core_model.Resource, ns string) *admission.Response {
-	if c.Mode == core.Global {
-		resourceOrigin, originPresent := core_model.ResourceOrigin(r.GetMeta())
-		if originPresent && resourceOrigin != mesh_proto.GlobalResourceOrigin {
-			return forbiddenResponse(labelsNotAllowedMsg(mesh_proto.ResourceOriginLabel, "global", string(resourceOrigin)))
-		}
-	}
-	if c.Mode != core.Global {
-		resourceOrigin, originPresent := core_model.ResourceOrigin(r.GetMeta())
-		if originPresent && resourceOrigin != mesh_proto.GlobalResourceOrigin {
-			zoneTag, ok := r.GetMeta().GetLabels()[mesh_proto.ZoneTag]
-			if ok && zoneTag != c.ZoneName {
-				return forbiddenResponse(labelsNotAllowedMsg(mesh_proto.ZoneTag, c.ZoneName, zoneTag))
+	if !c.DisableOriginLabelValidation {
+		switch c.Mode {
+		case core.Global:
+			resourceOrigin, originPresent := core_model.ResourceOrigin(r.GetMeta())
+			if !c.DisableOriginLabelValidation && originPresent && resourceOrigin != mesh_proto.GlobalResourceOrigin {
```

Comment:
at this point of the code there is no need to use `c.DisableOriginLabelValidation` (it's `false` since you're in this branch)